Model/Product/Image.php

```diff
@@ -74,7 +74,7 @@ private function isFastlyImageOptimizationEnabled()
             $this->isFastlyEnabled = false;
         }
 
-        if ($this->_scopeConfig->getValue(PageCacheConfig::XML_PAGECACHE_TYPE) !== Config::FASTLY) {
+        if ($this->_scopeConfig->getType() !== Config::FASTLY) {
```

Comment:
Currently, Magento isn't converting to int, but it will do so in the future.

```suggestion
        if ((int)$this->_scopeConfig->getType() !== Config::FASTLY) {
```

Reply:
@ihor-sviziev
I think you made a mistake, _scopeConfig is instanceof \Magento\Framework\App\Config\ScopeConfigInterface and not 
instance of \Fastly\Cdn\Model\Config
```suggestion
        if ((int) $this->_scopeConfig->getValue(PageCacheConfig::XML_PAGECACHE_TYPE) !== Config::FASTLY) {
```

Reply:
Thanks, both. Appreciate that.